src/components/RegisterUser/RegisterUser.test.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import RegisterUser from './RegisterUser'
+import { render } from '@testing-library/react'
+
+const childrenText = 'The Children'
+const children = <div>{childrenText}</div>
+
+describe('RegisterUser', () => {
```

Comment:
so for this component it felt much better to test the component without the connector. This is an unusually complex component (three possible branches!), and I still currently think the integration style is better for simpler components. But this has reduced my confidence in that a little.

For now I think we just keep doing what seems right in a given case, and check back in regularly to see if there's a conclusion to come to.